presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java

```diff
@@ -346,14 +347,19 @@ protected RowExpression visitLambdaExpression(LambdaExpression node, Void contex
         @Override
         protected RowExpression visitBindExpression(BindExpression node, Void context)
         {
-            RowExpression value = process(node.getValue(), context);
-            RowExpression function = process(node.getFunction(), context);
+            List<RowExpression> arguments = Stream.concat(
+                    node.getValues().stream()
+                            .map(value -> process(value, context)),
+                    Stream.of(process(node.getFunction(), context)))
+                    .collect(toImmutableList());
+            List<Type> argumentTypes = arguments.stream()
+                    .map(RowExpression::getType)
+                    .collect(toImmutableList());
```

Comment:
I would use a for-loop here to populate two `ImmutableList.Builder` here. I find it pretty hard to read this way.